oetube/aspnet-core/OeTube/Domain/Services/PlaylistItemManager.cs

```diff
@@ -0,0 +1,40 @@
+﻿using OeTube.Domain.Entities.Playlists;
+using OeTube.Domain.Entities.Videos;
+using OeTube.Domain.Repositories;
+using OeTube.Domain.Repositories.Extensions;
+using OeTube.Entities;
+using Volo.Abp.DependencyInjection;
+using Volo.Abp.Domain.Entities;
+using Volo.Abp.Domain.Services;
+
+namespace OeTube.Domain.Services
+{
+    public class PlaylistItemManager:DomainService
+    {
+        private readonly IVideoRepository _videoRepository;
+        private readonly IPlaylistRepository _playlistRepository;
+
+        public PlaylistItemManager(IVideoRepository videoRepository, IPlaylistRepository playlistRepository)
+        {
+            _videoRepository = videoRepository;
+            this._playlistRepository = playlistRepository;
+        }
+
+        public async Task<Playlist> UpdateMembersAsync(Playlist playlist, IEnumerable<Guid> items, bool autoSave = false, CancellationToken cancellationToken = default)
+        {
+            items = items.Distinct();
+            var videos=await _videoRepository.GetManyAsSetAsync(items, false, cancellationToken);
+            if (videos.Count != items.Count())
+            {
+                throw new EntityNotFoundException(typeof(Video));
+            }
+            if(videos.Any(v=>v.Id!=playlist.CreatorId))
+            {
+                throw new InvalidOperationException("Video and playlist creators do not match.");
```

Comment:
Is this a requirement? Without this it would be possible to add someone else's video to a playlist right?

Reply:
It would be possible, and Kovi also stated that only one's own video can be added to the playlist.